src/Validation/Validator.php

```diff
@@ -556,9 +573,13 @@ protected function _processRules($field, ValidationSet $rules, $data, $newRecord
                 continue;
             }
 
-            $errors[$name] = $message;
             if (is_string($result)) {
                 $errors[$name] = $result;
+            } elseif ($this->_useI18n) {
+                $args = $rule->get('pass');
+                $errors[$name] = __d($this->_validationDomain, $name, $this->_translateArgs($args));
```

Comment:
@lorenzo The error messages are already indexed by rule name so nothing more needs to be done in that respect. 

Repeating the message each time is a PITA for me personally. If you can suggest a way to easily achieve this through just my app code I will gladly close this PR.